File: tasks/3.2.1/3.2.1.py
def find_more_half(arr):
    map = {}
    for i in range (len(arr)):
        if (arr[i] in map):
            map[arr[i]] += 1
            if (map[arr[i]] > len(arr) / 2):
                return arr[i]
        else:
            map[arr[i]] = 1
    return -1

n = int(input())
arr = list(map(int, input().split()))
print(find_more_half(arr))

File: tasks/3.2.1/3.2.1.ts
function findMoreHalf(arr: number[]): number {
    const arrMap = new Map<number, number>()
    for (const item of arr) {
        const currentCount = arrMap.get(item) ?? 0
        if (currentCount + 1 > arr.length / 2) {
            return item
        } else {
            arrMap.set(item, currentCount + 1)
        }
    }
    return -1
}

console.log(findMoreHalf([7, 7, 8, 8, 8, 8, 9]))